src/client/dfs/dfs.c

```diff
@@ -949,12 +949,87 @@ entry_stat(dfs_t *dfs, daos_handle_t th, daos_handle_t oh, const char *name, siz
 
 	switch (entry.mode & S_IFMT) {
 	case S_IFDIR:
+	{
+		daos_handle_t	dir_oh;
+		daos_epoch_t	ep;
+
 		size = sizeof(entry);
-		stbuf->st_mtim.tv_sec = entry.mtime;
-		stbuf->st_mtim.tv_nsec = entry.mtime_nano;
-		stbuf->st_ctim.tv_sec = entry.ctime;
-		stbuf->st_ctim.tv_nsec = entry.ctime_nano;
+
+		/** check if dir is empty */
+		rc = daos_obj_open(dfs->coh, entry.oid, DAOS_OO_RO, &dir_oh, NULL);
+		if (rc) {
+			D_ERROR("daos_obj_open() Failed, "DF_RC"\n", DP_RC(rc));
+			return daos_der2errno(rc);
+		}
+
+		rc = daos_obj_query_max_epoch(dir_oh, th, &ep, NULL);
+		if (rc) {
+			daos_obj_close(oh, NULL);
+			return daos_der2errno(rc);
+		}
```

Comment:
Do we expect a performance impact of added RPCs?

Reply:
yes of course. but don't expect to see it in your tests currently since you do not stat directories in mdtest easy/hard (since we run those with -F)